bliss/sleep.py

```diff
@@ -342,31 +342,33 @@ def get_metrics(self, batch):
         true_params = {k: v for k, v in batch.items() if k not in exclude}
         true_params = get_full_params(slen, true_params)
 
-        # to compute metrics at the end.
-        counts_acc = 0.0
-        galaxy_counts_acc = 0.0
-        mses = []
-        for i in range(batch_size):
-            image = images[None, i]
-            true_n_sources_i = true_params["n_sources"][i].item()
-            true_n_galaxies_i = true_params["galaxy_bool"][i].sum().item()
+        # get map estimates
+        estimates = self.image_encoder.map_estimate(images)
 
-            estimate = self.image_encoder.map_estimate(image)
-            n_sources_i = estimate["n_sources"].item()
-            n_galaxies_i = estimate["galaxy_bool"].sum().item()
+        # accuracy of counts
+        counts_acc = (true_params["n_sources"] == estimates["n_sources"]).float().mean()
 
-            # calculate accuracies
-            counts_acc += (true_n_sources_i == n_sources_i) / batch_size
-            galaxy_counts_acc += (true_n_galaxies_i == n_galaxies_i) / batch_size
+        # acuraccy of galaxy counts
+        est_n_gal = estimates["galaxy_bool"].view(batch_size, -1).sum(-1)
+        true_n_gal = estimates["galaxy_bool"].view(batch_size, -1).sum(-1)
+        galaxy_counts_acc = (est_n_gal == true_n_gal).float().mean()
+
+        # accuracy of locations
+        est_locs = estimates["locs"]
+        true_locs = true_params["locs"]
+
+        mses = []
+        for i in range(batch_size):
```

Comment:
I think this for loop is OK for the locations, later we can improve it if necessary.